cmd/updates_test.go

```diff
@@ -0,0 +1,292 @@
+			assert.ElementsMatch(t, tt.validArchs, repo.Archs)
+		})
+	}
+}
+
+func TestProcWebChunk(t *testing.T) {
+	tests := []struct {
+		name       string
+		maint      string
+		product    string
+		validArchs []string
+		want       []get.HTTPRepoConfig
+		netWorkErr bool
+		wantErr    bool
+	}{
+		{
+			"Valid maint repo - single valid arch", "http://download.suse.de/ibs/SUSE:/Maintenance:/2/", "SUSE_SLE-15-SP4_Update/",
+			[]string{"aarch64"},
+			[]get.HTTPRepoConfig{
+				{
+					URL:   "http://download.suse.de/ibs/SUSE:/Maintenance:/2/SUSE_SLE-15-SP4_Update/",
+					Archs: []string{"aarch64"},
+				},
+			},
+			false,
+			false,
+		},
+		{
+			"Valid maint repo - multiple valid archs", "http://download.suse.de/ibs/SUSE:/Maintenance:/3/", "SUSE_SLE-15-SP4_Update/",
+			[]string{"x86_64", "aarch64", "ppc64le", "s390x"},
+			[]get.HTTPRepoConfig{
+				{
+					URL:   "http://download.suse.de/ibs/SUSE:/Maintenance:/3/SUSE_SLE-15-SP4_Update/",
+					Archs: []string{"x86_64", "aarch64", "ppc64le", "s390x"},
+				},
+			},
+			false,
+			false,
+		},
+		{
+			"Valid maint repo - no valid archs", "http://download.suse.de/ibs/SUSE:/Maintenance:/4/", "SUSE_SLE-15-SP4_Update/",
+			[]string{},
+			[]get.HTTPRepoConfig{},
+			false,
+			true,
+		},
+		{
+			"Network error", "http://download.suse.de/ibs/SUSE:/Maintenance:/5/", "SUSE_SLE-15-SP4_Update/",
+			[]string{},
+			[]get.HTTPRepoConfig{},
+			true,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := createMockClient(tt.maint, tt.validArchs, tt.netWorkErr)
+
+			got, err := ProcWebChunk(client, tt.product, tt.maint)
+			assert.EqualValues(t, tt.wantErr, (err != nil))
+			assert.Equal(t, len(tt.want), len(got))
+			for i := range tt.want {
+				wantRepo := tt.want[i]
+				gotRepo := got[i]
+				assert.Equal(t, wantRepo.URL, gotRepo.URL)
+				assert.ElementsMatch(t, wantRepo.Archs, gotRepo.Archs)
+			}
+		})
+	}
+}
+
+func TestGetProductsForMU(t *testing.T) {
+	tests := []struct {
+		name    string
+		mu      string
+		want    []string
+		wantErr bool
+	}{
+		{
+			"Chunk without 'SUSE' is discarded",
+			"http://download.suse.de/ibs/SUSE:/Maintenance:/6/",
+			[]string{"SUSE_SLE-15-SP4_Update/", "SUSE_SLE-15-SP6_Update/"},
+			false,
+		},
+		{
+			"Network error",
+			"http://download.suse.de/ibs/SUSE:/Maintenance:/7/",
+			nil,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client := createMockClient(tt.mu, []string{}, tt.wantErr)
+
+			got, err := getProductsForMU(client, tt.mu)
+			assert.EqualValues(t, tt.wantErr, (err != nil))
+			assert.ElementsMatch(t, tt.want, got)
+		})
+	}
+}
+
+func TestGetRepo(t *testing.T) {
+	tests := []struct {
+		name       string
+		mu         string
+		validArchs []string
+		want       []get.HTTPRepoConfig
+		netWorkErr bool
+		wantErr    bool
+	}{
+		{
+			"Single arch", "http://download.suse.de/ibs/SUSE:/Maintenance:/8/",
```

Comment:
As this tool is part of uyuni-project, it might be better to use OBS urls.

Reply:
Makes sense but the feature/command actually it  works only with IBS (even though there's an OBS package, I know), and I'd like to not give any false impression.

I'll have another PR (hopefully soon) to address packages naming and structure.